File: agent/src/main/java/com/smancode/smanagent/smancode/core/ParamsFormatter.java
package com.smancode.smanagent.smancode.core;

import java.util.Map;

/**
 * 参数格式化器
 * <p>
 * 提供参数的格式化功能
 */
public final class ParamsFormatter {

    private ParamsFormatter() {
        // 工具类，不允许实例化
    }

    /**
     * 格式化参数简述
     *
     * @param params 参数 Map
     * @return 格式化后的参数字符串
     */
    public static String formatBrief(Map<String, Object> params) {
        if (params == null || params.isEmpty()) {
            return "无";
        }

        StringBuilder sb = new StringBuilder();
        sb.append("{");
        params.forEach((key, value) -> {
            if (sb.length() > 1) {
                sb.append(", ");
            }
            String valueStr = String.valueOf(value);
            if (valueStr.length() > 50) {
                valueStr = valueStr.substring(0, 50) + "...";
            }
            sb.append(key).append("=").append(valueStr);
        });
        sb.append("}");
        return sb.toString();
    }
}

File: src/main/kotlin/com/smancode/sman/smancode/core/ParamsFormatter.kt
package com.smancode.sman.smancode.core

/**
 * 参数格式化器
 *
 * 提供参数的格式化功能
 */
object ParamsFormatter {

    /**
     * 格式化参数简述
     *
     * @param params 参数 Map
     * @return 格式化后的参数字符串
     */
    fun formatBrief(params: Map<String, Any>?): String {
        if (params.isNullOrEmpty()) {
            return "无"
        }

        val sb = StringBuilder()
        sb.append("{")
        params.forEach { (key, value) ->
            if (sb.length > 1) {
                sb.append(", ")
            }
            var valueStr = value.toString()
            if (valueStr.length > 50) {
                valueStr = valueStr.take(50) + "..."
            }
            sb.append(key).append("=").append(valueStr)
        }
        sb.append("}")
        return sb.toString()
    }
}
